org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/catalog/CatalogUtils.java

```diff
@@ -72,15 +83,99 @@ public static DOMAttr getCatalogEntryURI(DOMElement element) {
 	}
 
 	/**
-	 * Checks if this node is a catalog entry that is required to have the 'uri'
+	 * Returns the catalog attribute node of the given catalog entry or null if
+	 * there is no catalog attribute
+	 *
+	 * @param element The catalog entry to get the catalog attribute of
+	 * @return the catalog attribute node of the given catalog entry or null if
+	 *         there is no catalog attribute
+	 */
+	public static DOMAttr getCatalogEntryCatalog(DOMElement element) {
+		return element.getAttributeNode(CATALOG_ATTRIBUTE_NAME);
+	}
+
+	/**
+	 * Checks if this element is a catalog entry that is required to have the 'uri'
 	 * attribute
 	 *
-	 * @param node The node to check
-	 * @return true if this node is an catalog entry that is required to have the
+	 * @param element The element to check
+	 * @return true if this element is an catalog entry that is required to have the
 	 *         'uri' attribute and false otherwise
 	 */
-	private static boolean isCatalogURIEntry(DOMNode node) {
-		return node.isElement() && CATALOG_NAMES.contains(node.getNodeName());
+	private static boolean isCatalogEntryWithURI(DOMElement element) {
+		return HAS_URI_ATTRIBUTE.contains(element.getNodeName());
+	}
+
+	/**
+	 * Checks if this element is a catalog entry that is required to have the
+	 * 'catalog' attribute
+	 *
+	 * @param element The element to check
+	 * @return true if this element is an catalog entry that is required to have the
+	 *         'catalog' attribute and false otherwise
+	 */
+	private static boolean isCatalogEntryWithCatalog(DOMElement element) {
+		return HAS_CATALOG_ATTRIBUTE.contains(element.getNodeName());
+	}
+
+	/**
+	 *
+	 * @param element The element to check
+	 * @return true if the given element is a group element and false otherwise
+	 */
+	private static boolean isGroupCatalogEntry(DOMElement element) {
+		return GROUP_ENTITY_NAME.equals(element.getNodeName());
+	}
+
+	/**
+	 * Get a list of all catalog entry elements that are in the given catalog
+	 *
+	 * @param catalog the catalog element to collect the entries for
+	 * @return A list of all the catalog entities in this catalog
+	 */
+	private static List<CatalogEntry> collectCatalogEntries(@NonNull DOMElement catalog) {
+		List<CatalogEntry> entries = new ArrayList<>();
+		String baseURI = catalog.getAttribute(XML_BASE_ATTRIBUTE);
+		baseURI = baseURI == null ? "" : baseURI;
+		for (DOMNode node : catalog.getChildren()) {
+			if (node.isElement()) {
+				DOMElement element = (DOMElement) node;
+				if (isCatalogEntryWithURI(element)) {
```

Comment:
I would move this code in a private static method `createCatalogEntry` to use it too in `collectGroupEntries`

```java
DOMElement element = (DOMElement) node;
CatalogEntry entry = createCatalogEntry(element);
if (entry != null) {
    entries.add(entry);
}
```